htdocs/api/class/api_setup.class.php

```diff
@@ -1426,7 +1426,7 @@ public function getConf($confname)
     		throw new RestException(403, 'Error API open to admin users only or to the login user defined with constant API_LOGIN_ALLOWED_FOR_ADMIN_CHECK');
     	}
 
-    	if (! preg_match('/[^a-zA-Z0-9_]/', $confname) || ! isset($conf->global->$confname)) {
+    	if (! preg_match('/[a-zA-Z0-9_]/', $confname) || ! isset($conf->global->$confname)) {
```

Comment:
Test is here to be sure that confname contains only a-zA-Z0-9_ so test should be
if (! preg_match('/^[a-zA-Z0-9_]+$/', $confname)

Reply:
@eldy your fix is ok, we can found constante with _